### ISSUE TITLE
Koios 1.0.8: Asset object lacks consistency in different endpoints

### ISSUE THREAD

**Issue by AlexDochioiu** [cardano-community/koios-artifacts#108]
Going through the API docs, the `asset` object returned from `Address Information` is different from the one returned by `Account Assets`. Also one of them returns `asset_name` as a nullable hex, the other as a non-nullable ASCII. I don't see the purpose of those differences and it only makes it more confusing to use the API.

**Comment by dostrelith678** [comment]
`Address Information` will return the UTXO set of the address, which can have assets under the `asset_list` field. We could consider renaming that field to simply `assets`.

I presume you prefer the format where assets are grouped by policy ID and have an ASCI name field (i.e. `Account Assets`)?
```
"assets": [
      {
        "policy_id": "042afe60735e5915bf2d846314b0cf2072b880d6d4db947524173692",
        "assets": [
          {
            "asset_name": "45706f636873456e64333130",
            "asset_name_ascii": "EpochsEnd310",
            "balance": "1"
          }
        ]
      },
      ...
```

**Comment by AlexDochioiu** [comment]
I don't particularly care which one stays as long as all endpoints are consistent in returning the same kind of `asset` object. Also probably is redundant to include asset_name in both hex and ascii. Conversion can be done locally.

**Comment by rdlrt** [comment]
Thanks for raising, that's good feedback. Currently Account assets and Address Assets are aligned (share same schema), but not Address Assets vs Address Info (or Tx Info/UTxO). We should make them consistent

Just adding examples to the thread for easy reference:

1. Address Info

``` bash
curl -s -X POST "https://api.koios.rest/api/v0/address_info?select=utxo_set->0->asset_list"  -H "Accept: application/json"  -H "Content-Type: application/json"  -d '{"_addresses":["addr1qyswntr806udyffhf5ey05cnz0qjsnnvapp3nqc9qgmjfr07wwazuj0y0pndjydpgarxshe3xhfa0yhtv7t5c207f6tsjmwuuy"]}' | jq .
```
``` json
[
  {
    "asset_list": [
      {
        "policy_id": "750900e4999ebe0d58f19b634768ba25e525aaf12403bfe8fe130501",
        "asset_name": "424f4f4b",
        "quantity": "458539034"
      },
      {
        "policy_id": "f0ff48bbb7bbe9d59a40f1ce90e9e9d0ff5002ec48f232b49ca0fb9a",
        "asset_name": "626f6f6b5f696f",
        "quantity": "1"
      }
    ]
  }
]
```

2. Account Assets (same schema as Address Assets):
``` bash
curl -s -X POST "https://api.koios.rest/api/v0/account_assets?select=assets"  -H "Accept: application/json"  -H "Content-Type: application/json"  -d '{"_stake_addresses":["stake1u8l88w3wf8j8sekezxs5w3ngtucnt57hjt4k096v98lya9cm62l7k"]}' | jq .
```
``` json
[
  {
    "assets": [
      {
        "policy_id": "750900e4999ebe0d58f19b634768ba25e525aaf12403bfe8fe130501",
        "assets": [
          {
            "asset_name": "424f4f4b",
            "asset_name_ascii": "BOOK",
            "balance": "458539034"
          }
        ]
      },
      {
        "policy_id": "f0ff48bbb7bbe9d59a40f1ce90e9e9d0ff5002ec48f232b49ca0fb9a",
        "assets": [
          {
            "asset_name": "626f6f6b5f696f",
            "asset_name_ascii": "book_io",
            "balance": "1"
          }
        ]
      }
    ]
  }
]
```

3. Transaction Utxo / Info:
``` bash
curl -s -X POST "https://api.koios.rest/api/v0/tx_utxos?select=outputs->0->asset_list" -H "Accept: application/json"  -H "Content-Type: application/json"  -d '{"_tx_hashes":["d92d79fdbe87ed5556b01cfef65f247ce00132d6dd676f87cdc86842216fe8ff"]}' | jq .
```

``` json
[
  {
    "asset_list": [
      {
        "policy_id": "750900e4999ebe0d58f19b634768ba25e525aaf12403bfe8fe130501",
        "asset_name": "424f4f4b",
        "fingerprint": "asset15v2ehvq4gtl25vnxfr4r6rk6fxp549elhp7729",
        "quantity": "458539034"
      },
      {
        "policy_id": "f0ff48bbb7bbe9d59a40f1ce90e9e9d0ff5002ec48f232b49ca0fb9a",
        "asset_name": "626f6f6b5f696f",
        "fingerprint": "asset1myfw37pklr0z43gwyz0vnh0kju69nu9nrr8wz2",
        "quantity": "1"
      }
    ]
  }
]
```

**Comment by AlexDochioiu** [comment]
@rdlrt Glad to hear you'd consider aligning them! Also, unrelated but I just noticed that the API specs are wrong for `account_assets` endpoint (tested on mainnet)

I get the following
```json
[
  {
    "stake_address": "stake1u9qenantzm8fadvynmvkgul6ecp9kt5me003udf26smznqg3lfz4a",
    "assets": [
      {
        "policy_id": "0a4352475d66381d5bc7257224725ea7a83a115ff81c257db07a683c",
        "assets": [{ "asset_name": "49446a6565666f", "asset_name_ascii": "IDjeefo", "balance": "1" }]
      },
      {
        "policy_id": "530a197fe7c275f204c3396b3782fc738f4968f0c81dd2291cf07b8a",
        "assets": [
          {
            "asset_name": "4343303030313030303030303030303030303031313430303332",
            "asset_name_ascii": "CC000100000000000001140032",
            "balance": "1"
          },
          {
            "asset_name": "4343303030333030303030303030303030303031313430303631",
            "asset_name_ascii": "CC000300000000000001140061",
            "balance": "1"
          }
        ]
      },
      {
        "policy_id": "a7904896a247d3aa09478e856769b82d1f2e060028b6bda5543b699f",
        "assets": [
          { "asset_name": "4343434f4c4c41423030303031", "asset_name_ascii": "CCCOLLAB00001", "balance": "1" },
          { "asset_name": "4343434f4c4c41423038383839", "asset_name_ascii": "CCCOLLAB08889", "balance": "1" },
          { "asset_name": "4343434f4c4c41423038383931", "asset_name_ascii": "CCCOLLAB08891", "balance": "1" },
          {
            "asset_name": "4375746543726561747572657343686164694e617373617230303031",
            "asset_name_ascii": "CuteCreaturesChadiNassar0001",
            "balance": "1"
          },
          {
            "asset_name": "4375746543726561747572657343686164694e617373617230303033",
            "asset_name_ascii": "CuteCreaturesChadiNassar0003",
            "balance": "1"
          }
        ]
      }
    ]
  }
]
```

the API spec for each asset indicates `asset_name`, `asset_policy` and `balance`. The response contains `asset_name_ascii` and does not contain `asset_policy`

<img width="839" alt="image" src="https://user-images.githubusercontent.com/38853913/195067585-2c9418fe-6132-4966-84b9-e58950888963.png">

**Comment by rdlrt** [comment]
Hey Alex,

Going through all the endpoints to align the asset objects, will include fix for asset information under account endpoints as well

**Comment by rdlrt** [comment]
Hey @AlexDochioiu , Could you put an eye over PR linked - to see if there are any other modifications you'd expect?

**Comment by rdlrt** [comment]
Closed via koios-1.0.9rc
